inspirehep/records/api/base.py

```diff
@@ -981,3 +1015,83 @@ def dumps_for_es(self):
     def get_value(self, field, default=None):
         """Method which makes ``get_value`` more intuitive"""
         return get_value(self, field, default)
+
+    def is_superseded(self):
+        """Checks if record is superseded
+
+        Returns:
+            bool: True if is superseded, False otherwise
+        """
+        if "successor" in self.get_value("related_records.relation", ""):
+            return True
+        return False
+
+    def _update_refs_in_citation_table(self, save_every=100):
+        """Updates all references in citation table.
+            First removes all references (where citer is this record),
+            then adds all from the record again.
+
+        Args:
+            save_every (int): How often data should be saved into session.
+                One by one is very inefficient, but also 10000 at once.
+        """
+        with db.session.begin_nested():
+            CitationTable.query.filter_by(CiterId=self.id).delete()
+
+        if (
+            self.is_superseded()
+            or self.get("deleted")
+            or self.pid_type not in ["lit", "dat"]
+        ):
+            # Record is not eligible to cite
+            return
+        records_pids = self.get_linked_pids_from_field("references.record")
+        records_data = self.get_records_ids_by_pids(records_pids)
+        referenced_records = []
```

Comment:
this should be a `set`, as it's used for membership testing, which is inefficient for `list`s.